### PR TITLE
Initial xgboost callback

### PR DESCRIPTION
**Pull request by lukas**
This is a super simple callback that makes it easy to track xgboost models.

### CODE REVIEW

File: wandb/xgboost/__init__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+Example usage:
+
+param_list = [("eta", 0.08), ("max_depth", 6), ("subsample", 0.8), ("colsample_bytree", 0.8), ("objective", "multi:softmax"), ("eval_metric", "merror"), ("alpha", 8), ("lambda", 2), ("num_class", 10)]
+config.update(dict(param_list))
+bst = xgb.train(param_list, d_train, callbacks=[wandb_callback()])
+'''
+
+import wandb
+import xgboost
```
Comment (vanpelt) [inline wandb/xgboost/__init__.py]: Why are you importing xgboost here?

File: wandb/xgboost/__init__.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+def wandb_callback():
```
Comment (vanpelt) [inline wandb/xgboost/__init__.py]: It probably makes sense to name this WandbCallback to be consistent with the rest of the callbacks.

File: tests/test_xgboost.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import pytest
+import numpy as np
+import xgboost as xgb
```
Comment (vanpelt) [inline tests/test_xgboost.py]: You'll need to add xgboost to requirements_dev.txt

Comment (lukas) [reply tests/test_xgboost.py]: makes sense

File: wandb/xgboost/__init__.py (hunk 2 above)
Comment (lukas) [reply wandb/xgboost/__init__.py]: I don't know about this: WandbCallback is a class - in this case I'm defining a function.  Do we really and to uppercase a class?  Seems kind of wild.

File: wandb/xgboost/__init__.py (hunk 1 above)
Comment (vanpelt) [reply wandb/xgboost/__init__.py]: You still need to commit this.

Comment (lukas) [reply wandb/xgboost/__init__.py]: fixed

**Pull request merged**